core/objects.go

```diff
@@ -228,30 +228,34 @@ func (r *Registration) MergeUpdate(input Registration) {
 // challenge, we just throw all the elements into one bucket,
 // together with the common metadata elements.
 type Challenge struct {
+	ID int `json:"id,omitempty" db:"id"`
+
+	AuthorizationID string `json:"authorizationID,omitempty" db:"authorizationID"`
+
 	// The type of challenge
-	Type string `json:"type"`
+	Type string `json:"type" db:"type"`
```

Comment:
:+1:, this will also allow us to move the `ID + AuthorizationID` to the model, along with the `LockCol` (good call, forgot about that), to the model and avoid having to blank them out before sending authorizations to the client to maintain spec compliance.